Require copyright and source notices in credits

kemitchell opened the pull request:
cc @jeremy-w

kemitchell committed:
Require copyright and source notices in credits

With thanks to Jeremy W. Sherman

jeremy-w commented on LICENSE.mustache:
```diff
@@ -29,8 +29,9 @@ latest contribution.
 contributed to this software, and indirectly to your software,
-and make sure that it is easy for users to find when using your
-software. For example, if users access your software via the
+including the texts of the copyright and source notices above.
```
Should this be “**by** including”?

kemitchell replied:
Might word it better, but my initial answer is _no_. Users need to do all of these:

1. Tell users I contributed to this software.
2. Tell users I contributed indirectly to their software.
3. Include my copyright and source notices.

I don't think just copies of the copyright and source notices effectively achieves credit. Credit for the open software needs to be linked to credit for the software users are actually using.

jeremy-w replied:
OK, then I definitely found the late-floating ing clause confusing. I didn’t parse this as “the three things you need to do”. I think partly it’s the “and” earlier confusing things.

I think it may be clearer to reorder as: “to give credit, include these things and give users of your software notice that I both contributed to this software and also indirectly to your software.”

Even still, I’m not terribly clear on what a Notices page that credits, say, 5 different projects using this license would look like in terms of making it a readable structure. Probably something like, “The following persons contributed indirectly to this software via their work on the listed software:

* Jane Doe, SomeLibrary, link to license text anchor further down the page goes here”

That is, frontload the in/direct credit bits and stuff the license details below the fold. So I guess I can figure it out once I know those three separate things are needed.

kemitchell replied:
I am about to push a couple more commits that I think help clarify the requirements.

As for your notes on what the actual credits page would look like, what you described matches what I had in mind.

kemitchell committed:
Clarify required credit content

kemitchell committed:
Use "my credit" throughout credit instructions

kemitchell commented:
@jeremy-w, here's from the latest commit:

> To give me credit, give users of your software notice that I contributed to this software, and indirectly to your software. Make it easy for users to find my credit when using your software, and include the texts of the copyright and source notices above. For example, if users access your software via the World Wide Web, put my credit at the bottom of the home page, or put a link there to a page with my credit in its main text.

Feel right?

jeremy-w commented:
That “, and indirectly” comma looks off to me, because that’s not an independent clause. Recommend striking comma. Maybe optionally introduce both…and construction to stress that notice has to be given of both contributions.

That’s just a nit: I now have a much clearer idea of my responsibilities as a consumer of a library with that license. 👍🏽 
--
Jeremy W. Sherman
http://jeremywsherman.com/

> On Jul 19, 2018, at 17:53, Kyle Mitchell <notifications@github.com> wrote:
> 
> @jeremy-w, here's from the latest commit:
> 
> To give me credit, give users of your software notice that I contributed to this software, and indirectly to your software. Make it easy for users to find my credit when using your software, and include the texts of the copyright and source notices above. For example, if users access your software via the World Wide Web, put my credit at the bottom of the home page, or put a link there to a page with my credit in its main text.
> 
> Feel right?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

kemitchell committed:
Use directly ... indirectly for parallelism

With thanks to Jeremy W. Sherman

https://github.com/berneout/posterity-public-license/pull/3#issuecomment-406432625

kemitchell committed:
Format with `par -w65`

kemitchell commented:
@jeremy-w: How about we do it with parallelism?

> To give me credit, give users of your software notice that I contributed directly to this software and indirectly to your software. Make it easy for users to find my credit when using your software, and include the texts of the copyright and source notices above. For example, if users access your software via the World Wide Web, put my credit at the bottom of the home page, or put a link there to a page with my credit in its main text.

jeremy-w commented:
Looks good to me.

--
Jeremy W. Sherman
http://jeremywsherman.com/

> On Jul 19, 2018, at 18:43, Kyle Mitchell <notifications@github.com> wrote:
> 
> @jeremy-w: How about we do it with parallelism?
> 
> To give me credit, give users of your software notice that I contributed directly to this software and indirectly to your software. Make it easy for users to find my credit when using your software, and include the texts of the copyright and source notices above. For example, if users access your software via the World Wide Web, put my credit at the bottom of the home page, or put a link there to a page with my credit in its main text.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
